### PR TITLE
[PDK-12] Initialize backend (Draft)

### PR DESCRIPTION
**Pull request by santhitak**
This PR will resolve: [Initialize backend #12](https://github.com/deeptech-kmitl/producktivity/issues/12)

### Objective of the changes:
- Add Nest.js to workspace
- Init backend for application

### Changes proposed:
- Workspace has Nest.js as BE

### Check list for reviewer:

- [x] Information in the description is enough and understandable.
- [x] Does not have performance issues.
- [x] Code changes does not effect build failed.
- [ ] Does not have issues.

**Approving the changes, reviewer must fully understands the description. DO NOT APPROVE MR IF YOU HAVE NO IDEA WHAT IS GOING ON**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,28 +33,39 @@
...
+    "@swc/cli": "^0.1.63",
     "@swc/core": "~1.3.102",
+    "@types/jest": "^29.4.0",
     "@types/node": "^20.10.6",
     "@typescript-eslint/eslint-plugin": "^6.17.0",
     "@typescript-eslint/parser": "^6.17.0",
     "@vite-pwa/assets-generator": "^0.2.2",
     "@vitest/ui": "~1.1.1",
     "autoprefixer": "~10.4.11",
+    "axios": "^1.0.0",
```
Comment (spicyzboss) [inline package.json:45]: we really use `axios` in backend?

File: package.json [hunk 2]
```diff
@@ -25,28 +33,39 @@
...
+    "jest": "^29.4.1",
+    "jest-environment-node": "^29.4.1",
     "jsonc-eslint-parser": "^2.4.0",
     "lightningcss": "^1.22.1",
     "lint-staged": "^15.2.0",
     "nx": "17.2.8",
     "postcss": "^8.4.32",
     "prettier": "^3.1.1",
     "qwik-nx": "^2.0.2",
+    "reflect-metadata": "^0.1.13",
```
Comment (spicyzboss) [inline package.json:60]: we don't need polyfill like this when we not use in older version of ts.

File: package.json [hunk 3]
```diff
@@ -25,28 +33,39 @@
...
+    "rxjs": "^7.8.0",
     "storybook-framework-qwik": "^0.2.5",
     "tailwindcss": "^3.4.0",
+    "ts-jest": "^29.1.0",
+    "ts-node": "10.9.1",
+    "tslib": "^2.3.0",
     "typescript": "~5.3.3",
+    "url-loader": "^4.1.1",
```
Comment (spicyzboss) [inline package.json:68]: this package is for webpack do we will use it?

File: package.json [hunk 4]
```diff
@@ -9,10 +9,18 @@
   "devDependencies": {
     "@builder.io/qwik": "~1.3.2",
     "@builder.io/qwik-city": "~1.3.2",
+    "@nestjs/common": "^10.0.2",
+    "@nestjs/core": "^10.0.2",
+    "@nestjs/platform-express": "^10.0.2",
```
Comment (spicyzboss) [inline package.json:14]: would be great if you migrate to Fastify by default, for the performance reason.

**Review by spicyzboss — CHANGES_REQUESTED** [review]
pls move the api project within `api/` group folder.

like this:
image

I'm also moving it inside group folder for more descriptive and more easy to manage.

File: package.json (hunk 3 above)
Comment (santhitak) [reply package.json:68]: no, we're not.

File: package.json (hunk 4 above)
Comment (santhitak) [reply package.json]: I have migrate from express to fastify, please help check changes again.

**Comment by santhitak** [comment]
> pls move the api project within `api/` group folder.
> 
> like this: image
> 
> I'm also moving it inside group folder for more descriptive and more easy to manage.

you meant like this? Not sure what structure do you want it to be, could you screenshot the whole of it?

```
/apps
--/api
 ---/api
 ---/api-e2e
```

**Comment by spicyzboss** [comment]
like the `apps/api/` is the descriptive of all api apps in the folder, then we named the child folder to the domain. like we will make authentication api, create the nestjs 
project named `auth` in folder `api/`. So, the project should be `api/auth/` and `api/auth-e2e`.

**Pull request merged**